PyRDF/backend/Dist.py

```diff
@@ -306,6 +315,10 @@ def mapper(current_range):
                 for f in current_range.filelist:
                     chain.Add(f)
 
+                if current_range.friend_filelist:
+                    for f in current_range.friend_filelist:
+                        chain.AddFriend(f[0], f[1])
```

Comment:
This assumes that the friend trees only have one file, but it may also be a chain composed by many files.

First we need to recreate the same chain before adding it as friend, as in the root implementation.

Additionally, we cannot just add the friend, since we need to set the number of entries to be read, as we do for `chain` with `chain.SetCacheEntryRange`.